### PR TITLE
Energy sword and dual sword QoL

### PR DESCRIPTION
**Pull request by MadmanMartian**
Idea courtesy of Durgen.

 - Makes dualsaber a subtype of energy sword. Weird that it wasn't before
 - Fixes esword combination logic. No longer can you combine a banana blade and a pirates saber into darth mauls lightsaber.
 - Combining the two blades into a dualsaber now puts the dualsaber in your hands
 - You can now deconstruct the dualsaber back into its composite blades. It attempts to put the blades into your hands afterwards.

:cl:
 * rscadd: Can now deconstruct dual sabers into its composite sabers. This requires a screwdriver.
 * tweak: Dualblade saber construction logic changed. Can now only use energy swords to make a dualblade, or two banana blades to make a bunch blade.
 * tweak:  Constructing a dualblade now puts it in your hands. Same goes for deconstructing the dualblade and putting the composite swords in your hand.

### CODE REVIEW

File: code/game/objects/items/weapons/melee/energy.dm [hunk 1]
```diff
@@ -143,22 +146,34 @@
...
+		user.drop_item(W)
+		W.forceMove(saber)
+		user.drop_item(src)
+		forceMove(saber)
+		user.put_in_hands(saber)
+		return 1
+	return 0
 
+/obj/item/weapon/melee/energy/sword/proc/can_combine_with(obj/item/W)
+	return istype(W, /obj/item/weapon/melee/energy/sword) && !is_type_in_list(W, list(/obj/item/weapon/melee/energy/sword/bsword, /obj/item/weapon/melee/energy/sword/pirate, /obj/item/weapon/melee/energy/sword/dualsaber))
```
Comment (boy2mantwicethefam) [inline code/game/objects/items/weapons/melee/energy.dm:168]: Is it possible to replace these with a var on the e-swords where if the var is equal to the other item's var you can combine them?

**Comment by SonixApache** [comment]
How do you deconstruct it?
Update the wiki with all this information, please.

**Comment by MadmanMartian** [comment]
It says in the changelog, and pointless to update the wiki if this isn't merged yet.

**Comment by jknpj** [comment]
can the same be done with machetes?

**Comment by SonixApache** [comment]
I thought machetes were intended to be permanently combined?

File: code/game/objects/items/weapons/melee/energy.dm [hunk 2]
```diff
@@ -192,6 +203,128 @@
...
+ * Double-Bladed Energy Swords
+ */
+
+/obj/item/weapon/melee/energy/sword/dualsaber
+	icon_state = "dualsaber0"
+	name = "double-bladed energy sword"
+	desc = "Handle with care."
+	var/colorset = ""
+	force = 3
+	throwforce = 5.0
```
Comment (Probe1) [inline code/game/objects/items/weapons/melee/energy.dm:219]: uh, why 5.0?  The .0 specifically.  I don't remember ever seeing a throwforce as a float and also why if it's going to be 0 anyway.  Obviously you're just doing a file cleanup here and didn't write it yourself but I'm intrigued about what the original coder was doing there.

**Review by Probe1 — APPROVED** [review]

**Pull request merged**